### PR TITLE
Little improvements

### PR DESCRIPTION
**Pull request by steven-murray**
This PR adds several little improvements that I came across when attempting to run H4C sims on ILIFU:

* Removes some redundant conversion functions (lm stuff)
* Adds some wiggle room in checking if a beam covers the full sky (basically, it was ensuring that the maximum az was greater than 2pi - delta_az, but floating point errors sometimes means it doesn't quite make this criterion but still effectively covers the sky).
* If the beam covers the sky but only goes to 2pi-delta_az instead of all the way to 2pi, instead of raising a warning it just puts az=0 at az=2pi. 
* Makes the check on infs/nans in A_s both faster and optional. Turns out that taking a sum then checking if its nan/inf is faster than doing `np.any(np.isinf())`. Also, now it only checks this the first time the beam is interpolated, and trusts that the rest is fine if it works the first time. This is not quite comprehensive, but is probably better than actually checking every time.

### CODE REVIEW

File: src/vis_cpu/cpu.py [hunk 1]
```diff
@@ -100,6 +100,7 @@ def _evaluate_beam_cpu(
     ty: np.ndarray,
     polarized: bool,
     freq: float,
+    check: bool = False,
```
Comment (r-pascua) [inline src/vis_cpu/cpu.py:103]: Add description for this parameter to docstring.

**Review by r-pascua — CHANGES_REQUESTED** [review]
Looks good to me! If we're only interpolating the beam to sources above the horizon, then I think we're safe only doing the check on the first integration (at least for realistic use cases).

Requested changes: remove the two obsolete tests (beam pixellization stuff), and figure out why the error-checking test is failing (and fix it).

**Review by r-pascua — APPROVED** [review]
Thanks for the improvements! Happy to have this merged, but please update the `_evaluate_beam_cpu` docstring with a description of what the `check` parameter does.

File: src/vis_cpu/cpu.py [hunk 2]
```diff
@@ -301,6 +305,7 @@ def vis_cpu(
             ty,
             polarized,
             freq,
+            check=t == 0,
```
Comment (piyanatk) [inline src/vis_cpu/cpu.py:308]: Is this a valid syntax or a typo?

**Review by piyanatk — APPROVED** [review]
Look good @steven-murray. Thank you for your hard work. I just have one comment on probably a syntax that I don't understand.

One general question, would it make sense to check for beam discontinuity at az=0 and 2pi.

File: src/vis_cpu/cpu.py (hunk 2 above)
Comment (steven-murray) [reply src/vis_cpu/cpu.py:308]: It's valid!

**Comment by steven-murray** [comment]
Thanks @piyanatk: I added a check for the discontinuity (which is also tested).
Thanks @r-pascua: the docs have been updated.

If tests pass, I'll merge!

**Pull request merged**